test/UnitTests/MSTest.Analyzers.UnitTests/PreferAssertFailOverAlwaysFalseConditionsAnalyzerTests.cs

```diff
@@ -1470,4 +1452,77 @@ public void TestMethod()
 
         await VerifyCS.VerifyCodeFixAsync(code, code);
     }
+
+    [TestMethod]
+    public async Task WhenIsNullAssertion_ReferenceTypeInNullableDisableContext_NoDiagnostic()
+    {
+        string code = """
+            using System.Diagnostics.CodeAnalysis;
+            using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+            [TestClass]
+            public sealed class MyTests
+            {
+            #nullable disable
+                [TestMethod]
+                public void Repro()
+                {
+                    Assert.IsFalse(TryGetPool(poolKey: null, out var pool));
+                    Assert.IsNull(pool);
+                }
+            #nullable restore
+
+            #nullable enable
+                internal bool TryGetPool(string? poolKey, [MaybeNullWhen(false)] out object pool)
+                {
+                    if (poolKey is not null)
+                    {
+                        pool = new object();
+                        return true;
+                    }
+
+                    pool = null;
+                    return false;
+                }
+            #nullable restore
+            }
+            """;
+
+        await VerifyCS.VerifyCodeFixAsync(code, code);
```

Comment:
Does this also verify that MSTEST0025 is not reported?
If not, I don't see a test for the scenario I described in the issue.

Reply:
Yes, `VerifyCodeFixAsync(code, code)` asserts that no analyzer diagnostic is reported.